src/base/bittorrent/session.cpp

```diff
@@ -3123,7 +3124,7 @@ void Session::handleIPFilterError()
 }
 
 #if LIBTORRENT_VERSION_NUM < 10100
-void Session::dispatchAlerts(std::auto_ptr<libt::alert> alertPtr)
+void Session::dispatchAlerts(std::unique_ptr<libt::alert> &alertPtr)
```

Comment:
Semantically this signature makes no sense, AFAIK: either `Session::dispatchAlerts()` takes ownership of the object, and then pass rvalue reference there, or it does not (not true), and then it does not need  `std::unique_ptr` at all. So, I suggest
```C++
void Session::dispatchAlerts(std::unique_ptr<libt::alert> &&alertPtr);
```
and 
```C++
dispatchAlerts(std::move(p));
```
above